core/src/main/java/org/axonframework/queryhandling/QueryGateway.java

```diff
@@ -0,0 +1,21 @@
+package org.axonframework.queryhandling;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+/**
```

Comment:
The same javadoc comments as earlier, but now also for all the implementable functions